pkg/controller/validation/ippool.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2025 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"fmt"
+	"net/netip"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	crdv1beta1 "antrea.io/antrea/pkg/apis/crd/v1beta1"
+)
+
+func GetIPRangeSet(ipRanges []crdv1beta1.IPRange) sets.Set[string] {
+	set := sets.New[string]()
+	for _, ipRange := range ipRanges {
+		ipRangeStr := ipRange.CIDR
+		if ipRangeStr == "" {
+			ipRangeStr = fmt.Sprintf("%s-%s", ipRange.Start, ipRange.End)
+		}
+		set.Insert(ipRangeStr)
+	}
+	return set
+}
+
+func ParseIPRangeCIDR(cidrStr string) (netip.Prefix, string) {
+	var cidr netip.Prefix
+	var err error
+
+	cidr, err = netip.ParsePrefix(cidrStr)
+	if err != nil {
+		return cidr, fmt.Sprintf("invalid cidr %s", cidrStr)
+	}
+	cidr = cidr.Masked()
+	return cidr, ""
+}
+
+func ParseIPRangeStartEnd(startStr, endStr string) (netip.Addr, netip.Addr, string) {
+	var start, end netip.Addr
+	var err error
+
+	start, err = netip.ParseAddr(startStr)
+	if err != nil {
+		return start, end, fmt.Sprintf("invalid start ip address %s", startStr)
+	}
+
+	end, err = netip.ParseAddr(endStr)
+	if err != nil {
+		return start, end, fmt.Sprintf("invalid end ip address %s", endStr)
+	}
+	return start, end, ""
+}
+
+func ValidateSubnetInfo(gateway string, prefixLength int32) error {
```

Comment:
There is nothing specific to the "subnet info" here, or to gateways. You are just validating an IP address with prefix.

I am not even sure this function is called anywhere, can you check?